rfcs/0001-object_info_v2.md

```diff
@@ -0,0 +1,304 @@
+    },
+    {
+        "type": "CONDITIONING",
+        "name": "negative",
+        "is_list": False,
+        "tooltip": "negative-tooltip"
+    }
+)
+```
+
+#### 1.2 Input Type Definitions
+
+Input types will be explicitly defined using tuples with configuration objects:
+
+```python
+@classmethod
+def INPUT_TYPES(s):
+    return {
+        "required": {
+            # File-based combos become FILE_COMBO type
+            "ckpt_name": ("FILE_COMBO", {
+                "folder_path": "checkpoints",
+                "filter_ext": [".ckpt", ".safetensors"]
+            }),
+            # Regular combos become explicit
+            "mode": ("COMBO", {
+                "options": ["balanced", "speed", "quality"],
+                "default": "balanced",
+                "tooltip": "Processing mode"
+            })
+        }
+    }
+```
+
+### Phase 2: API Changes
+
+#### 2.1 New Endpoints
+
+```python
+@routes.get("/api/v2/list_files/{folder_name}")
+async def list_folder_files(request):
+    folder_name = request.match_info["folder_name"]
+    filter_ext = request.query.get("filter_ext", "").split(",")
+
+    files = folder_paths.get_filename_list(folder_name)
+    if filter_ext and filter_ext[0]:
+        files = [f for f in files if any(f.endswith(ext) for ext in filter_ext)]
+
+    return web.json_response({
+        "files": files,
+        "folder": folder_name
+    })
+
+@routes.get("/api/v2/object_info")
+async def get_object_info(request):
+    node_class = request.query.get("node_class")
+    include_combos = request.query.get("include_combos", "false").lower() == "true"
+
+    response = {}
+    if node_class:
+        # Single node info
+        response[node_class] = node_info_v2(node_class, include_combos)
+    else:
+        # All nodes info (default behavior)
+        for cls in nodes.NODE_CLASS_MAPPINGS:
+            response[cls] = node_info_v2(cls, include_combos)
+
+    return web.json_response(response)
+```
+
+#### 2.2 New Response Format
+
+Old format:
+
+```javascript
+{
+    "CheckpointLoader": {
+        "input": {
+            "required": {
+                "ckpt_name": [[
+                    "file1",
+                    "file2",
+                    ...
+                    "fileN",
+                ]],
+                "combo_input": [[
+                    "option1",
+                    "option2",
+                    ...
+                    "optionN",
+                ], {
+                    "default": "option1",
+                    "tooltip": "Processing mode"
+                }],
+            },
+            "optional": {}
+        },
+        "output": ["MODEL"],
+        "output_name": ["model"],
+        "output_is_list": [false],
+        "output_tooltip": ["The loaded model"],
+        "output_node": false,
+        "category": "loaders"
+    }
+}
+```
+
+New format:
+
+```javascript
+{
+    "CheckpointLoader": {
+        "input": {
+            "required": {
+                "ckpt_name": [
```

Comment:
Here is a decision I am not very sure about. Whether we go one step further and flatten the structure here:

From
```python
"combo input": ["COMBO", { options: [1, 2, 3], default: 2}]
```
to
```python
"combo input":  {"type": "COMBO", "options": [1, 2, 3], "default": 2}
```

This shall further reduce the complexity parsing the API response, at a cost of an extra named field.